analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -205,31 +233,35 @@ class SpdxDocumentFile(
             Project.EMPTY.copy(id = Identifier.EMPTY.copy(type = managerName))
         }
 
-        val nonProjectPackages = if (projectPackage != null) {
-            spdxDocument.packages - projectPackage
-        } else {
-            spdxDocument.packages
+        if (projectPackage != null) {
+            packages.remove(makePackageFromSpdxPackage(projectPackage, workingDir))
```

Comment:
Is this really necessary? I  think the `projectPackage`  is never added. And before your changes this was only needed, because you got all packages from the `spdxDocument` which did not distinguish between project and non project packages.